### DISCUSSION TITLE
Show tides as a graph

### DISCUSSION THREAD

**Discussion by nicholaswaton** [Daveiano/weewx-wdc#235] [Ideas]
First thank you for all the work you've done on this wonderful skin for WeeWx, it looks pretty amazing. 

I am currently building a weather station page for my community sailing club.
https://weather.cal-sailing.org/index.html

One thing we care about a lot is the tides as it can influence when we can go sailing or not.

I was wondering if it might be possible to display the tides that get retrieved via weewx-forecast in graph format, either for the day, or for the week.
Ideally we would have a way to also set a threshold, which would be the minimum tide when we go sailing, anything below is too shallow (and could be shown in red).

**Comment by nicholaswaton** [comment]
What I am trying to do is like this graph: https://suiattle.net/CapeSmallHarborWeather/sp/bt/tidegraph/
With a set threshold line running through it (set at 1.5ft for us).

**Comment by Daveiano** [comment]
Hi @nicholaswaton 

thank you for reaching out! Sounds like an interesting use-case, I will have a look if that's possible - but I think it should be possible. Iam currently out of free time - so please be a little patient ;)

**Comment by Daveiano** [comment]
So I finally got some time to research this!

Bad news first, from the creator of weewx-forecast:

> forecast uses a completely different type of database.  the data are not directly plotable, so imagegenerator cannot use the forecast database.  timestamps are not unique keys in the forecast database. [...]

https://groups.google.com/g/weewx-development/c/IbwZdEjY0ec/m/dsN3Z7ttXVQJ

So it seems that would need some code from the skin itself to transform the data to support plotting it. Since that's a bigger task than I thought, I would like to put this feature into the 3.6.0 release (not the upcoming 3.5.0). That is also the case for #236

FYI: I will keep the "Task" labels on these two discussions for tracking, once I am working on the 3.6.0 release, I will transfer these into issues.
